src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -762,6 +796,81 @@ internal static void ValidateSpecifiedAuthentication(PSCredential credential, st
 
         #endregion
 
+        #region Internal Methods
+
+        /// <summary>
+        /// Parse the Connection parameter HashTable array.
+        /// </summary>
+        /// <returns>Array of SSHConnection objects</returns>
+        private const string ComputerNameParameter = "ComputerName";
+        private const string HostNameAlias = "HostName";
+        private const string UserNameParameter = "UserName";
+        private const string KeyFilePathParameter = "KeyFilePath";
+        private const string IdentityFilePathAlias = "IdentityFilePath";
+        internal SSHConnection[] ParseSSHConnectionHashTable()
+        {
+            List<SSHConnection> connections = new List<SSHConnection>();
+            foreach (var item in this.SSHConnection)
+            {
+                if (item.ContainsKey(ComputerNameParameter) && item.ContainsKey(HostNameAlias))
+                {
+                    throw new PSArgumentException(RemotingErrorIdStrings.SSHConnectionDuplicateHostName);
+                }
+
+                if (item.ContainsKey(KeyFilePathParameter) && item.ContainsKey(IdentityFilePathAlias))
+                {
+                    throw new PSArgumentException(RemotingErrorIdStrings.SSHConnectionDuplicateKeyPath);
+                }
+
+                SSHConnection connectionInfo = new SSHConnection();
+                foreach (var key in item.Keys)
+                {
+                    string paramName = key as string;
+                    if (string.IsNullOrEmpty(paramName))
+                    {
+                        throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
+                    }
+
+                    string paramValue = item[paramName] as string;
+                    if (string.IsNullOrEmpty(paramValue))
+                    {
+                        throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
+                    }
+
+                    if (paramName.Equals(ComputerNameParameter, StringComparison.OrdinalIgnoreCase) || paramName.Equals(HostNameAlias, StringComparison.OrdinalIgnoreCase))
+                    {
+                        var resolvedComputerName = ResolveComputerName(paramValue);
+                        ValidateComputerName(new string[] { resolvedComputerName });
+                        connectionInfo.ComputerName = resolvedComputerName;
+                    }
+                    else if (paramName.Equals(UserNameParameter, StringComparison.OrdinalIgnoreCase))
+                    {
+                        connectionInfo.UserName = paramValue;
+                    }
+                    else if (paramName.Equals(KeyFilePathParameter, StringComparison.OrdinalIgnoreCase) || paramName.Equals(IdentityFilePathAlias, StringComparison.OrdinalIgnoreCase))
```

Comment:
After talking offline I think this is Ok as is.  This parameter set is intended to support fan-out remoting where it is possible that 100s of target machines are specified in the hashtable.  But I feel the parsing of these connection info objects is insignificant compared to setting up the remote runspaces needed for each connection.  But this is something we can revisit if it turns out to be a perf problem.